keras/objectives.py

```diff
@@ -49,11 +49,9 @@ def poisson(y_true, y_pred):
 
 
 def cosine_proximity(y_true, y_pred):
-    assert K.ndim(y_true) == 2
-    assert K.ndim(y_pred) == 2
-    y_true = K.l2_normalize(y_true, axis=1)
-    y_pred = K.l2_normalize(y_pred, axis=1)
-    return -K.mean(y_true * y_pred, axis=1)
+    y_true = K.l2_normalize(y_true, axis=-1)
+    y_pred = K.l2_normalize(y_pred, axis=-1)
+    return -K.mean(y_true * y_pred, axis=-1)
```

Comment:
No I'm not, it just came up while I was writing tests that it didn't support time-distributed and I figured there was no downside to adding it and making it consistent with the other objectives.